packages/account-tree-controller/src/rules/entropy.ts

```diff
@@ -0,0 +1,111 @@
+import type { Bip44Account } from '@metamask/account-api';
+import {
+  AccountWalletCategory,
+  isBip44Account,
+  toMultichainAccountId,
+  toMultichainAccountWalletId,
+} from '@metamask/account-api';
+import { isEvmAccountType, type EntropySourceId } from '@metamask/keyring-api';
+import type { KeyringControllerState } from '@metamask/keyring-controller';
+import type { InternalAccount } from '@metamask/keyring-internal-api';
+
+import { Rule } from './rule';
+import type { AccountTreeGroup, AccountTreeWallet } from '..';
+import type { AccountTreeControllerMessenger } from '../AccountTreeController';
+import type { AccountContext } from '../types';
+
+/**
+ * Select keyrings from keyring controller state.
+ *
+ * @param state - The keyring controller state.
+ * @returns The keyrings.
+ */
+function selectKeyringControllerKeyrings(state: KeyringControllerState) {
+  return state.keyrings;
+}
+
+export class EntropyRule extends Rule {
+  readonly category = AccountWalletCategory.Entropy;
+
+  readonly #entropySourcesToIndex: Map<EntropySourceId, number>;
+
+  constructor(messenger: AccountTreeControllerMessenger) {
+    super(messenger);
+
+    this.#entropySourcesToIndex = new Map();
+    this.#syncEntropySources();
```

Comment:
Actually, this might be a mistake. Keyrings might not be available at that time (if extension is still locked).

I'll move this to a runtime call to solve that issue.

Reply:
Ended-up making a direct call everytime. That feels a bit overkill to listen to the `KeyringController` state change anyway. Much more concise that way.